lib/api/2.0/obms.js

```diff
@@ -45,13 +46,56 @@ var obmsGetById = controller(function(req) {
 
 // PATCH /api/2.0/obms/identifier
 var obmsPatchById = controller( function(req) {
-    return waterline.obms.updateByIdentifier(
-        req.swagger.params.identifier.value, req.swagger.params.body.value);
+    var obmId = req.swagger.params.identifier.value;
+    var values = req.swagger.params.body.value;
+    return waterline.obms.needByIdentifier(obmId)
+    .then(function(oldObm) {
+        if (oldObm.node && !values.nodeId) {
+            return Promise.all([waterline.nodes.getNodeById(oldObm.node)]);
+        } else if (!oldObm.node && values.nodeId) {
+            return Promise.all([waterline.nodes.getNodeById(values.node)]);
+        } else if (oldObm.node && values.nodeId && oldObm.node === values.nodeId) {
+            return Promise.all([waterline.nodes.getNodeById(oldObm.node)]);
+        } else if (oldObm.node && values.nodeId && oldObm.node !== values.nodeId) {
+            return Promise.all([waterline.nodes.getNodeById(oldObm.node),
+                    waterline.nodes.getNodeById(values.nodeId)]);
+        }
+    })
+    .then(function(oldNodes) {
+        return waterline.obms.updateByIdentifier(obmId, values)
+        .tap(function() {
+            _.forEach(oldNodes, function(oldNode) {
+                if (oldNode) {
+                    /* asynchronous */
```

Comment:
Please add more comment about why you intentionally want the promise not be returned.